examples/sui-move/dynamic_fields/sources/example.move

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module dynamic_fields::example {
+    use sui::dynamic_object_field as ofield;
+    use sui::object::{Self, UID};
+    use sui::test_scenario;
+
+    struct Parent has key {
+        id: UID,
+    }
+
+    struct Child has key, store {
+        id: UID,
+        count: u64,
+    }
+
+    public fun add_child(parent: &mut Parent, child: Child) {
+        ofield::add(&mut parent.id, b"child", child);
+    }
+
+    public fun mutate_child(child: &mut Child) {
```

Comment:
Honestly I don't understand this example.

Reply:
Which part?  There's no deep meaning to this function, it's just a function that modifies an instance of an object type.

Reply:
Yeah, but in this context it feels like you can update the Child in two ways.

Reply:
You can, because it's possible to `reclaim_child` and then modify it directly, but I can add a doc comment explaining that this is only callable when `child` is not a dynamic field of a `parent`.